app/main/views/frameworks.py

```diff
@@ -230,7 +230,7 @@ def get_framework_suppliers(framework_slug):
         agreement_returned = request.args.get('agreement_returned')
         if agreement_returned is not None:
             if convert_to_boolean(agreement_returned):
-                requested_statuses = requested_statuses or ('signed', 'on-hold', 'countersigned')
+                requested_statuses = requested_statuses or ('signed', 'on-hold', 'approved', 'countersigned')
```

Comment:
Side note there's an awful lot of python stuff going on around here that is probably better done in sql.